### PR TITLE
chore(contributing): update description of modules in project

### LINKED ISSUE #2504: Update contributer guide modules overview to current state
raft, gossip etc are gone, zb-db has emerged

### PR DESCRIPTION
**Pull request by jwulf**
closes #2504 

* Remove `raft` and `gossip`
* Add `db`, `gateway` and `gateway-protocol`

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -20,8 +20,9 @@ This is a small overview of the contents of the different modules:
 - `service-container` is a custom implementation to manage dependencies between different services
 - `logstreams` is an implementation of a append only log backed by the filesystem
 - `transport` is our abstraction over network transports
-- `gossip` is a cluster membership and failure detection implementation
-- `raft` is a cluster consensus and replication implementation
+- `gateway` is the implementation of the gRPC gateway, using our SBE-based protocol to communicate with brokers
+- `gateway-protocol` is the gRPC definitions for the Zeebe client-to-gateway protocol
+- `db` is our RocksDB wrapper for state management
```
Comment (Zelldon) [inline CONTRIBUTING.md]:
```suggestion
- `zb-db` is our RocksDB wrapper for state management
```

**Review by ChrisKujawa — DISMISSED** [review]
Thanks @jwulf for updating this. Probably we should also add the engine module.

File: CONTRIBUTING.md (hunk 1 above)
Comment (jwulf) [reply CONTRIBUTING.md]: Updated and pushed.

**Review by ChrisKujawa — APPROVED** [review]
bors r+

**Pull request merged**